rpa-mapper/src/main/java/uk/gov/hmcts/cmc/rpa/mapper/RequestForJudgementJsonMapper.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.cmc.rpa.mapper;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.CountyCourtJudgment;
+import uk.gov.hmcts.cmc.domain.models.ccj.PaymentOption;
+import uk.gov.hmcts.cmc.domain.models.ccj.RepaymentPlan;
+import uk.gov.hmcts.cmc.rpa.DateFormatter;
+import uk.gov.hmcts.cmc.rpa.mapper.json.NullAwareJsonObjectBuilder;
+
+import javax.json.JsonObject;
+
+@Component
+public class RequestForJudgementJsonMapper {
+
+    public JsonObject map(Claim claim) {
+        CountyCourtJudgment countyCourtJudgment = claim.getCountyCourtJudgment();
+        return new NullAwareJsonObjectBuilder()
+            .add("caseNumber", claim.getReferenceNumber())
+            .add("ccjRequestedOn", DateFormatter.format(claim.getCountyCourtJudgmentRequestedAt()))
+            .add("amountWithInterest", claim.getTotalAmountTillToday().orElse(null))
+            .add("courtFee", claim.getClaimData().getFeesPaidInPound())
+            .add("alreadyPaid", countyCourtJudgment.getPaidAmount().orElse(null))
+            .add("payment", getPaymentValue(countyCourtJudgment))
+            .add("instalmentAmount", countyCourtJudgment.getRepaymentPlan()
+                .map(RepaymentPlan::getInstalmentAmount).orElse(null))
+            .add("firstPayment", countyCourtJudgment.getRepaymentPlan()
```

Comment:
Can we please have something like this?

```
{
  "paymentType": "" // IMMEDIATELY / BY SET DATE / INSTALMENTS
  "fullPaymentDeadline": "",
  "instalments": {
    "amount": "",
    "firstPayment": "",
    "frequency": "" // EACH WEAK / EVERY TO WEEKS / MONTHLY
  }
}
```

I mean structure that is easier to understand then current, flattened one?

Reply:
Refactored it to get two switch values together

Reply:
Actioned....finally :)